app/src/main/java/com/onegini/mobile/exampleapp/view/activity/PendingPushMessagesActivity.java

```diff
@@ -0,0 +1,175 @@
+import com.onegini.mobile.sdk.android.handlers.OneginiPendingMobileAuthWithPushRequestsHandler;
+import com.onegini.mobile.sdk.android.handlers.error.OneginiPendingMobileAuthWithPushRequestError;
+import com.onegini.mobile.sdk.android.model.entity.OneginiMobileAuthWithPushRequest;
+import com.onegini.mobile.sdk.android.model.entity.UserProfile;
+
+public class PendingPushMessagesActivity extends AppCompatActivity {
+
+
+  @SuppressWarnings({ "unused", "WeakerAccess" })
+  @BindView(R.id.toolbar)
+  Toolbar toolbar;
+  @SuppressWarnings({ "unused", "WeakerAccess" })
+  @BindView(R.id.recycler_view)
+  RecyclerView recyclerView;
+  @BindView(R.id.pending_notifications_error)
+  TextView errorTextView;
+  @BindView(R.id.bottom_navigation)
+  BottomNavigationView bottomNavigationView;
+  @BindView(R.id.notifications_refresh_layout)
+  SwipeRefreshLayout swipeRefreshLayout;
+
+  private PendingPushMessagesAdapter adapter;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_pending_notifications);
+    ButterKnife.bind(this);
+    setupUi();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    fetchPendingTransactions();
+  }
+
+  private void setupUi() {
+    setupActionBar();
+    setupListView();
+    setupNavigationBar();
+  }
+
+  private void setupListView() {
+    adapter = new PendingPushMessagesAdapter(this);
+    recyclerView.setAdapter(adapter);
+    recyclerView.setLayoutManager(new LinearLayoutManager(this));
+    swipeRefreshLayout.setOnRefreshListener(() -> fetchPendingTransactions());
+  }
+
+  private void setupNavigationBar() {
+    bottomNavigationView.getMenu().findItem(R.id.action_notifications).setChecked(true);
+    bottomNavigationView.setOnNavigationItemSelectedListener(item -> {
+      final int itemId = item.getItemId();
+      if (itemId == R.id.action_resources) {
+        onHttpClicked();
+      } else if (itemId == R.id.action_notifications) {
+        onNotificationsClicked();
+      } else {
+        onHomeClicked();
+      }
+      return true;
+    });
+  }
+
+  private void fetchPendingTransactions() {
+    swipeRefreshLayout.setRefreshing(true);
+    recyclerView.setVisibility(View.GONE);
+    errorTextView.setVisibility(View.GONE);
+    OneginiSDK.getOneginiClient(this).getUserClient().getPendingMobileAuthWithPushRequests(new OneginiPendingMobileAuthWithPushRequestsHandler() {
+      @Override
+      public void onSuccess(final Set<OneginiMobileAuthWithPushRequest> set) {
+        swipeRefreshLayout.setRefreshing(false);
+        displayFetchedMessages(set);
+      }
+
+      @Override
+      public void onError(final OneginiPendingMobileAuthWithPushRequestError oneginiPendingMobileAuthWithPushRequestError) {
+        swipeRefreshLayout.setRefreshing(false);
+        showError(oneginiPendingMobileAuthWithPushRequestError.getMessage());
```

Comment:
It would have been nice to display the error code as well for the future integration tests. There's ErrorMessageParser.class that can help with that.

Reply:
OK, fixed